Enable XATTR_USER_PREFIX extended attribute namespace for tmpfs

For XATTR_PAX_FLAGS markings to work on a tmpfs filesystem, we
need to accept XATTR_USER_PREFIX extended attribute namespace
as valid.  In Gentoo and other distros that make use of tmpfs
for their packaging systems, this makes it possible to pax mark
executables built in tmpfs before being tarred or otherwised
packaged.

X-Gentoo-Bug: 432434
X-Gentoo-Bug-URL: https://bugs.gentoo.org/show_bug.cgi?id=432434
Signed-off-by: Anthony G. Basile <blueness@gentoo.org>
---

diff --git a/mm/shmem.c b/mm/shmem.c
index 67afba5..697a181 100644
--- a/mm/shmem.c
+++ b/mm/shmem.c
@@ -2208,7 +2208,8 @@ static int shmem_xattr_validate(const char *name)
 {
 	struct { const char *prefix; size_t len; } arr[] = {
 		{ XATTR_SECURITY_PREFIX, XATTR_SECURITY_PREFIX_LEN },
-		{ XATTR_TRUSTED_PREFIX, XATTR_TRUSTED_PREFIX_LEN }
+		{ XATTR_TRUSTED_PREFIX, XATTR_TRUSTED_PREFIX_LEN },
+		{ XATTR_USER_PREFIX, XATTR_USER_PREFIX_LEN }
 	};
 	int i;
 
